### PR TITLE
MRG, ENH: Treat MEG channel type EXCI as 'MEGOTHER'

### PR DESCRIPTION
**Pull request by dnacombo**
PR Description
--------------

Fixing issue in utils.py/_get_ch_type_mapping.
Channel type EXCI (from neuromag system) does not have a mapping.
Adding it to "MEGOTHER" (?)

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [x] All comments resolved
- [x] This is not your own PR
- [ ] All CIs are happy
- [x] PR title starts with [MRG]
- [x] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/master/doc/whats_new.rst) is updated

### CODE REVIEW

**Comment by agramfort** [comment]
can you add a test along these lines : https://github.com/mne-tools/mne-bids/blob/master/mne_bids/tests/test_write.py#L685 ?

**Review by sappelhoff — COMMENTED** [review]
lgtm (adding a line for a test as Alex suggested would be welcome).

Just out of curiosity: what kind of data does the EXCI channel contain? What is EXCI an abbreviation for?

**Comment by dnacombo** [comment]
This is how it looks:
image

**Comment by agramfort** [comment]
@dnacombo I think you have flake8 issues

File: mne_bids/tests/test_write.py [hunk 1]
```diff
@@ -698,6 +698,21 @@ def test_fif_ias(tmpdir):
...
+    raw = _read_raw_fif(raw_fname)
+
+    raw.set_channel_types({raw.ch_names[0]: 'exci'})
+
+    data_path = BIDSPath(subject='sample', root=tmpdir)
+
+    write_raw_bids(raw, data_path)
+    raw = read_raw_bids(data_path)
+    assert raw.info['chs'][0]['kind'] == FIFF.FIFFV_EXCI_CH
+
```
Comment (sappelhoff) [inline mne_bids/tests/test_write.py:715]:
```suggestion

def test_fif_exci(tmpdir):
    """Test writing FIF files with excitation channel."""
    data_path = testing.data_path()
    raw_fname = op.join(data_path, 'MEG', 'sample',
                        'sample_audvis_trunc_raw.fif')
    raw = _read_raw_fif(raw_fname)

    raw.set_channel_types({raw.ch_names[0]: 'exci'})

    data_path = BIDSPath(subject='sample', root=tmpdir)

    write_raw_bids(raw, data_path)
    raw = read_raw_bids(data_path)
    assert raw.info['chs'][0]['kind'] == FIFF.FIFFV_EXCI_CH

```

**Comment by hoechenberger** [comment]
Exci-ting!!! 😁

**Comment by dnacombo** [comment]
> @dnacombo I think you have flake8 issues

I hope it's not contagious!

**Comment by sappelhoff** [comment]
> I hope it's not contagious!

I tried treating you with https://github.com/mne-tools/mne-bids/pull/728/commits/ce7f0973a20111e586af0b956b25324984676967, but I added some new error of my own :)

I think some `with pytest.raises(RuntimeWarning) ...` is needed

**Comment by dnacombo** [comment]
> I think some `with pytest.raises(RuntimeWarning) ...` is needed

That's beyond my competence. I didn't expect a blank line would do such thing...

File: mne_bids/tests/test_write.py [hunk 2]
```diff
@@ -699,6 +699,22 @@ def test_fif_ias(tmpdir):
...
 
+def test_fif_exci(tmpdir):
+    """Test writing FIF files with excitation channel."""
+    data_path = testing.data_path()
+    raw_fname = op.join(data_path, 'MEG', 'sample',
+                        'sample_audvis_trunc_raw.fif')
+    raw = _read_raw_fif(raw_fname)
+
+    raw.set_channel_types({raw.ch_names[0]: 'exci'})
+
```
Comment (sappelhoff) [inline mne_bids/tests/test_write.py:710]:
```suggestion
    match = 'The unit for channel(s) MEG 0113 has changed'
    with pytest.raises(RuntimeError, match=match):
        raw.set_channel_types({raw.ch_names[0]: 'exci'})
```

**Comment by sappelhoff** [comment]
You can see my suggestion in https://github.com/mne-tools/mne-bids/pull/728/files#r594944982

basically, your code raises a "RuntimeWarning" from MNE to alert the potential user about a change of unit.

When testing with pytest, such warnings result in an error - to make sure nothing unexpected passes the tests.

So to tell pytest that we do indeed expect this warning, we need to use a "context manager" (here: `with pytest.warns`) with a regular expression (`match`) that matches the warning that we expect to occur.

If exactly one warning is raised by the code and it matches our regexp, the test passes - because everything is in order.

**Comment by sappelhoff** [comment]
> but I added some new error of my own :)

strictly speaking, what I said there was wrong --> the issue was there before I added the blank line (to fix another issue) ... but it wasn't caught before due to the script erroring out before encountering it.

**Comment by agramfort** [comment]
+1 for MRG when CIs are green

File: mne_bids/tests/test_write.py [hunk 3]
```diff
@@ -699,6 +699,23 @@ def test_fif_ias(tmpdir):
...
 
+def test_fif_exci(tmpdir):
+    """Test writing FIF files with excitation channel."""
+    data_path = testing.data_path()
+    raw_fname = op.join(data_path, 'MEG', 'sample',
+                        'sample_audvis_trunc_raw.fif')
+    raw = _read_raw_fif(raw_fname)
+
+    match = r'The unit for channel\(s\) MEG 0113 has changed'
+    with pytest.warns(RuntimeError, match=match):
```
Comment (sappelhoff) [inline mne_bids/tests/test_write.py:710]:
```suggestion
    with pytest.warns(RuntimeWarning, match=match):
```

File: mne_bids/tests/test_write.py [hunk 4]
```diff
@@ -699,6 +699,23 @@ def test_fif_ias(tmpdir):
     assert raw.info['chs'][0]['kind'] == FIFF.FIFFV_IAS_CH
 
 
+def test_fif_exci(tmpdir):
```
Comment (hoechenberger) [inline mne_bids/tests/test_write.py:702]:
```suggestion
@pytest.mark.filterwarnings(warning_str['channel_unit_changed'])
def test_fif_exci(tmpdir):
```

**Review by hoechenberger — APPROVED** [review]

**Review by agramfort — APPROVED** [review]

File: mne_bids/tests/test_write.py [hunk 5]
```diff
@@ -699,6 +699,24 @@ def test_fif_ias(tmpdir):
...
+    """Test writing FIF files with excitation channel."""
+    data_path = testing.data_path()
+    raw_fname = op.join(data_path, 'MEG', 'sample',
+                        'sample_audvis_trunc_raw.fif')
+    raw = _read_raw_fif(raw_fname)
+
+    match = r'The unit for channel\(s\) MEG 0113 has changed'
+    with pytest.warns(RuntimeWarning, match=match):
+        raw.set_channel_types({raw.ch_names[0]: 'exci'})
```
Comment (sappelhoff) [inline mne_bids/tests/test_write.py:712]: I think this part isn't needed anymore now, right @hoechenberger ?

```suggestion
    raw.set_channel_types({raw.ch_names[0]: 'exci'})
```

**Comment by hoechenberger** [comment]
Thanks @dnacombo!

**Pull request merged**